crates/host/src/wasmbus/mod.rs

```diff
@@ -0,0 +1,642 @@
+            .map_err(|e| anyhow!(e.to_string()).context("failed to lock"))
+    }
+}
+
+impl Queue {
+    #[instrument]
+    async fn new(
+        nats: &async_nats::Client,
+        cluster_key: &KeyPair,
+        host_key: &KeyPair,
+    ) -> anyhow::Result<Self> {
+        let (registries, pings, links, queries, auction, commands, inventory) = try_join!(
+            nats.subscribe("wasmbus.ctl.default.registries.put".into()),
+            nats.subscribe("wasmbus.ctl.default.ping.hosts".into()),
+            nats.subscribe("wasmbus.ctl.default.linkdefs.*".into()),
+            nats.subscribe("wasmbus.ctl.default.get.*".into()),
+            nats.subscribe("wasmbus.ctl.default.auction.>".into()),
+            nats.subscribe(format!(
+                "wasmbus.ctl.default.cmd.{}.*",
+                host_key.public_key()
+            )),
+            nats.subscribe(format!(
+                "wasmbus.ctl.default.get.{}.inv",
+                host_key.public_key()
+            )),
+        )
+        .context("failed to subscribe to queues")?;
+        Ok(Self {
+            auction,
+            commands,
+            pings,
+            inventory,
+            links,
+            queries,
+            registries,
+        })
+    }
+}
+
+/// Wasmbus lattice
+#[derive(Debug)]
+pub struct Lattice {
+    event_builder: EventBuilderV10,
+    friendly_name: String,
+    host_key: KeyPair,
+    labels: HashMap<String, String>,
+    nats: async_nats::Client,
+    start_at: Instant,
+}
+
+impl Lattice {
+    const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(30);
+
+    /// Construct a new [Lattice] returning a tuple of it's [Arc] and an async shutdown function.
+    #[instrument]
+    pub async fn new(
+        LatticeConfig {
+            url,
+            cluster_seed,
+            host_seed,
+        }: LatticeConfig,
+    ) -> anyhow::Result<(Arc<Self>, impl Future<Output = anyhow::Result<()>>)> {
+        let cluster_key = if let Some(cluster_seed) = cluster_seed {
+            let kp = KeyPair::from_seed(&cluster_seed)
+                .context("failed to construct key pair from seed")?;
+            ensure!(kp.key_pair_type() == KeyPairType::Cluster);
+            kp
+        } else {
+            KeyPair::new(KeyPairType::Cluster)
+        };
+        let host_key = if let Some(host_seed) = host_seed {
+            let kp =
+                KeyPair::from_seed(&host_seed).context("failed to construct key pair from seed")?;
+            ensure!(kp.key_pair_type() == KeyPairType::Server);
+            kp
+        } else {
+            KeyPair::new(KeyPairType::Server)
+        };
+
+        let labels = HashMap::from([
+            ("hostcore.arch".into(), ARCH.into()),
+            ("hostcore.os".into(), OS.into()),
+            ("hostcore.osfamily".into(), FAMILY.into()),
+            (
+                "path".into(),
+                env::var("WASMCLOUD_HOST_PATH").unwrap_or_default(),
+            ),
```

Comment:
Is this just for debugging purposes? Or to overwrite the nix store PATH variable?

Reply:
By default, the only labels that I have on a vanilla OTP host is this:
```
  hostcore.osfamily                     unix
  hostcore.os                           macos
  hostcore.arch                         aarch64
```

Reply:
So I think unless you wanted to patch that bug with the `HOST_PATH` environment variable, you do not need the `path` label